src/overlay.js

```diff
@@ -105,29 +111,35 @@
             options.location.x,
             options.location.y,
             options.location.width,
-            options.location.height
-        );
-        this.position   = new $.Point(
-            options.location.x,
-            options.location.y
-        );
-        this.size       = new $.Point(
-            options.location.width,
-            options.location.height
-        );
-        this.style      = options.element.style;
-        // rects are always top-left
-        this.placement  = options.location instanceof $.Point ?
+            options.location.height);
+
+        // this.position is never read by this class but is kept for backward
+        // compatibility
+        this.position = this.bounds.getTopLeft();
+
+        // this.size is only used by PointOverlay with options.checkResize === false
+        this.size = this.bounds.getSize();
+
+        this.style = options.element.style;
+
+        // rects are always top-left (RectOverlays don't use placement)
```

Comment:
That makes sense however that doesn't work well with a Rectangle as location.
Maybe those overlays scalable in only one direction should take a Point location and a width (or height).
If we go that route, would that make sense to always have a point as location and scale width/height if those parameters are presents? Rectangles would still be accepted as syntactic sugar around specifying the top left location + width + height.